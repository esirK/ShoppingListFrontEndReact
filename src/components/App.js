import React, { Component } from 'react';
import RegisterForm from '../containers/register_form'

export default class App extends Component {
  render() {
    return (
      <div className="container">
        <RegisterForm />
      </div>
    );
  }
}