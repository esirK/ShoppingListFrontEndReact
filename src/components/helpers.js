export function validate(values){
	const errors = {};
	if (!values.username){
		errors.username = 'Username Required';
	}
	if (!values.email){
		errors.email = 'Email Required';
	}
	if (!values.password){
		errors.password = 'Password Required';
	}
	if (values.password && values.password.length < 6){
		errors.password = 'Passwords must be at least 6 Characters Long';
	}
	if (!values.confirm_password){
		errors.confirm_password = 'Enter confirmation Password';
	}
	if(values.password !== values.confirm_password){
		console.log('Look at me ',values.confirm_password);
		errors.confirm_password = 'Passwords do not match';
	}
	return errors;
}
