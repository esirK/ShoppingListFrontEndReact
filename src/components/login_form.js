import React, {Component} from 'react';


export default class LoginForm extends Component{
    render(){
        return(
            <div>
                <p>I am Mr Meesix Look at me</p>
            </div>
        );
    }
}