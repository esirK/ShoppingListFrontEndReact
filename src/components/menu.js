import React, {Component} from 'react';
import {connect} from 'react-redux';

import MenuItem from 'material-ui/MenuItem';
import TextField from 'material-ui/TextField';
import {FlatButton} from 'material-ui';

import Dialog from 'material-ui/Dialog';
import Snackbar from 'material-ui/Snackbar';

import IconMenu from 'material-ui/IconMenu';
import IconButton from 'material-ui/IconButton';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';

import {Logout, update_profile, hideSnackBar} from '../actions';

export class Menu extends Component{
	constructor(props){
		super(props);
		this.state ={
			conf_logout: false,
			open_profile: false,
			username: '',
			password: '',
			old_password: '',
			openSb: false
		};
		this.logout = this.logout.bind(this);
		this.profile = this.profile.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleLogout = this.handleLogout.bind(this);
		this.handlePasswordChange = this.handlePasswordChange.bind(this);
		this.handleUsernameChange = this.handleUsernameChange.bind(this);
		this.handleProfileUpdate = this.handleProfileUpdate.bind(this);
	}
	logout(){
		this.setState({conf_logout: true});
	}
	profile(){
		this.setState({open_profile: true});
		this.props.hideSnackBar();
	}
	handleLogout(){
		this.props.Logout();
		this.setState({conf_logout: false});
		//reload page on Logout
		window.location.reload();
	}
	handlePasswordChange(e){
		this.setState({password: e.target.value});
	}
	handleUsernameChange(e){
		this.setState({username: e.target.value});
	}
	handleProfileUpdate(){
		this.props.update_profile({username:this.state.username, password:this.state.password});
		this.setState({open_profile: false});
		this.setState({'username':''});
		this.setState({'password':''});
		this.setState({'old_password':''});
	}
	handleClose(){
		this.setState({conf_logout: false});
		this.setState({open_profile: false});
		this.setState({'username':''});
		this.setState({'password':''});
		this.setState({'old_password':''});
		this.props.hideSnackBar();
	}
	render(){
		let actions = [
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.handleClose}
			/>,
			<FlatButton
				label="OK"
				primary={true}
				onClick={this.handleLogout}
			/>,
		];
		let profile_actions = [
			<FlatButton
				label="Cancel"
				primary={true}
				onClick={this.handleClose}
			/>,
			<FlatButton
				label="Update"
				primary={true}
				onClick={this.handleProfileUpdate}
			/>,
		];
		const profile_children = [
			<TextField
			  hintText="New Username"
			  value={this.state.username}
			  fullWidth={true}
			  key='name'
			  onChange={this.handleUsernameChange}
			/>,
			<TextField
			  hintText="Old Password"
			  multiLine={true}
			  fullWidth={true}
			  value={this.state.password}
			  key='old_p'
			  onChange={this.handlePasswordChange}
			  type={'password'}
			/>,
			<TextField
			  hintText="New Password"
			  multiLine={true}
			  fullWidth={true}
			  key='new_p'
			  onChange={this.handlePasswordChange}
			  type="password"
			>
			</TextField>
		];
		return(
			<div>
				<IconMenu
					iconButtonElement={
						<IconButton><MoreVertIcon /></IconButton>
					}
					targetOrigin={{horizontal: 'left', vertical: 'top'}}
					anchorOrigin={{horizontal: 'right', vertical: 'top'}}
				>
					<MenuItem primaryText="Profile" onClick={this.profile} />
					<MenuItem primaryText="Sign out" onClick={this.logout}/>
				</IconMenu>
				<Dialog
					actions={actions}
					modal={false}
					open={this.state.conf_logout}
					onRequestClose={this.handleClose}
				>
         			Confirm Logout?
				</Dialog>
				<Dialog open={this.state.open_profile}
					title="Update Personal details"
					actions={profile_actions}
					children={profile_children}
					onRequestClose={this.handleClose}
				>
				</Dialog>
				<Snackbar
					open={this.props.openSb}
					message={this.props.message}
					autoHideDuration={3000}
					onRequestClose={this.handleRequestClose}
				/>
			</div>
		);
	}
}
function mapStateToProps(state) {
	return{
		message: state.authentication.message,
		openSb: state.authentication.openSb,
	};
}
export default connect(mapStateToProps, {Logout, update_profile, hideSnackBar})(Menu);