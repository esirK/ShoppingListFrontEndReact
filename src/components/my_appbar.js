import React, { Component } from 'react';

import {connect} from 'react-redux';
import AppBar from 'material-ui/AppBar';

import Menu from './menu';

export class MyAppBar extends Component{
	render(){
		return(
			<AppBar title="Shopping List App" showMenuIconButton={false}
				iconElementRight={this.props.isAuthenticated?<Menu />:null}
			/>
		);
	};
}
function mapStateToProps(state, ownProps) {
	return{
		isAuthenticated: state.authentication.isAuthenticated,
	};
}
export default connect(mapStateToProps, null)( MyAppBar);