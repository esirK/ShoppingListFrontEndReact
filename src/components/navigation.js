import React, { Component } from 'react';
import { Link } from 'react-router-dom';


export default class Navigation extends Component{
	render(){
		return(
			<div className="container">
				<div className="navbar-header">
					<a className="navbar-brand" href="/" title="Awesome Shopping list app">

					</a>
				</div>
				<div className="navbar-collapse ">
					<ul>
						<li>
							<Link to="/">
                                Home
							</Link>
						</li>
					</ul>
				</div>
			</div>
		);
	}
}