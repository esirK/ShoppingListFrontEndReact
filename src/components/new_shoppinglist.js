import React, {Component} from 'react';
import {connect} from 'react-redux';

import Dialog from 'material-ui/Dialog';
import CircularProgress from 'material-ui/CircularProgress';


import {itemActions, newShoppinglistChildren} from './helpers';
import {addNewShoppingList, resetErrors, closeFab} from '../actions';

import ShoppingLists from './shoppinglists';
export class AddShoppingList extends Component{

	constructor(props){
		super(props);
		this.state = {
			name: '',
			description: '',
			exit: false,
			open: true,
		};
		this.handleClose = this.handleClose.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
		this.handleTitleChange = this.handleTitleChange.bind(this);
		this.handleDescChange = this.handleDescChange.bind(this);
	}
	handleClose(){
		// resets state on closeing of Add shoppinglist dialog
		this.setState({ name: '' });
		this.setState({ description: ''});
		this.setState({ exit: true });
		this.props.closeFab();
		this.props.resetErrors();
	}
	handleSubmit(){
		/*Submits user request to create 
		a new shoppinglist with the required parameters*/
		let limit = this.props.limit;
		this.props.addNewShoppingList({
			'name': this.state.name,
			'description': this.state.description
		}, limit);
		this.setState({ name: '' });
		this.setState({ description: ''});
	}
	handleTitleChange(e){
		// Updates title field as user types
		this.setState({ name: e.target.value });
	}
	handleDescChange(e){
		// Updates description as user types
		this.setState({ description: e.target.value });
	}
	render()
	{
		const actions = itemActions(this);

		if(this.props.isCreatingNewShoppingList)
		{
			//add a circular progress bar bettween the 2 buttons
			actions.splice(1, 0, <CircularProgress/>);
		}
		const children = newShoppinglistChildren(this);
		//if the shoppinglist has been created then return to parent
		if(this.props.shoppinglistCreated | this.state.exit){
			return(
				<ShoppingLists/>
			);  
		}
		return(
			<Dialog open={this.state.open}
				title="Add Shoppinglist"
				actions={actions}
				children={children}
			>
			</Dialog>
		);
	};
}
function mapStateToProps(state){
	return {
		isCreatingNewShoppingList: state.shoppinglists.isCreatingNewShoppingList,
		error: state.shoppinglists.error,
		message: state.shoppinglists.message,
		shoppinglistCreated: state.shoppinglists.shoppinglistCreated,
	};
}
export default connect(mapStateToProps, {addNewShoppingList, closeFab,
	resetErrors}) (AddShoppingList);