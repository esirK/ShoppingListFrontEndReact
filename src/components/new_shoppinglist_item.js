import React, {Component} from 'react';

import Dialog from 'material-ui/Dialog';
import TextField from 'material-ui/TextField';

import {connect} from 'react-redux';

import {deactivateAddItem, viewShoppingList, addNewShoppingListItem} from '../actions';
import {itemActions} from './helpers';


export class NewShoppingListItem extends Component{
	constructor(props){
		super(props);
		this.state = {
			name: '',
			price: '',
			quantity: '',
			shoppinglist_id: -1
		};
		this.handleNameChange = this.handleNameChange.bind(this);
		this.handlePriceChange = this.handlePriceChange.bind(this);
		this.handleQuantityChange = this.handleQuantityChange.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}
	handleNameChange(e){
		//Update the name field
		this.setState({ name: e.target.value });
	}
	handlePriceChange(e){
		//Update the Price field
		this.setState({ price: e.target.value });
	}
	handleQuantityChange(e){
		//Update the Quantity field
		this.setState({ quantity: e.target.value });
	}
	
	handleClose(){
		//Invoked when the cancle button is clicked
		//Sets the openAddItem props to false
		this.props.deactivateAddItem();
	}
	handleSubmit(){
		//Invoked When a User presses the create new shoppinglist item button
		//submit details Supplied
		this.props.addNewShoppingListItem({
			'name': this.state.name,
			'price': this.state.price,
			'quantity': this.state.quantity,
			'shoppinglist_id': this.props.id
		});
		//reload the new shoppinglists
		//Update API to return all shoppinglist items on addition 
		//of a new one
		this.props.viewShoppingList(this.props.match.params.id);
	}
	render(){
		const actions = itemActions(this);
		let nameError = '';
		let priceError = '';
		let quantityError = '';
		let serverError='';
		if(this.props.error.name !== undefined){
			nameError=this.props.error.name[0].message;
		}
		if(this.props.error.price !== undefined){
			priceError=this.props.error.price[0].errors.Price;
		}
		if(this.props.error.quantity !== undefined){
			quantityError=this.props.error.quantity[0].errors.Quantity;
		}
		//server error
		if(this.props.error !== undefined){
			serverError=this.props.error;
		}
		let error = '';
		if(nameError){
			error=nameError;
		}else if(typeof(serverError) !== 'object'){
			error=serverError;
		}
		const children = [
			<TextField
			  hintText="Shoppinglist Item Name"
			  value={this.state.name}
			  fullWidth={true}
			  errorText={error}
			  onChange={this.handleNameChange}
			  key='name'
			/>,
			<TextField
			  hintText="Price of the Item"
			  multiLine={true}
			  fullWidth={true}
			  errorText={priceError}
			  value={this.state.price}
			  onChange={this.handlePriceChange}
			  key='price'
			/>,
			<TextField
			  hintText="Quantity of the Item"
			  multiLine={true}
			  fullWidth={true}
			  errorText={quantityError}
			  value={this.state.quantity}
			  onChange={this.handleQuantityChange}
			  key='quantity'
			/>
		];
		return(
			<Dialog open={this.props.openAddItem}
				title="Add Shoppinglist Item"
				actions={actions}
				children={children}
			>
			</Dialog>
		);
	}
}
function mapStateToProps(state){
	return{
		openAddItem: state.shoppinglist_items.openAddItem,
		id: state.shoppinglist_items.id,
	};
}
export default connect(mapStateToProps, {deactivateAddItem, viewShoppingList, addNewShoppingListItem})(NewShoppingListItem);