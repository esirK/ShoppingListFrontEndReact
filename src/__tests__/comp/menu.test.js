import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {Menu} from '../../components/menu';

describe('Render Menu from Update profile and Logout',()=>{
	const props = {
		openSb:false,
		message:''
	};
    
	const menu = shallow(<Menu {...props}/>);
	it('Renders Menu component', ()=>{
		expect(shallowToJson(menu)).toMatchSnapshot();
	});
});