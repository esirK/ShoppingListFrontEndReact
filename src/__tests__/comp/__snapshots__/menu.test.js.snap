// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Menu from Update profile and Logout Renders Menu component 1`] = `
<div>
  <IconMenu
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    animated={true}
    iconButtonElement={
      <IconButton
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <NavigationMoreVert />
      </IconButton>
    }
    multiple={false}
    onClick={[Function]}
    onItemTouchTap={[Function]}
    onKeyboardFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onRequestChange={[Function]}
    open={null}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    touchTapCloseDelay={200}
    useLayerForClickAway={false}
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      onClick={[Function]}
      primaryText="Profile"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      onClick={[Function]}
      primaryText="Sign out"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </IconMenu>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="OK"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
  >
    Confirm Logout?
  </Dialog>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Update"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Update Personal details"
  >
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      fullWidth={true}
      hintText="New Username"
      key="name"
      multiLine={false}
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value=""
    />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      fullWidth={true}
      hintText="New Password"
      key="new_p"
      multiLine={true}
      onChange={[Function]}
      rows={1}
      type="password"
      underlineShow={true}
    />
  </Dialog>
  <Snackbar
    autoHideDuration={3000}
    message=""
    open={false}
  />
</div>
`;
